### PR TITLE
Hermes: replace default JSON with human-readable output

### PR DESCRIPTION
**Pull request by adizere**
Closes: #805

Many thanks @romac for code snippets & helping with navigating the code around abscissa!

## Description

Primary change in this PR:

- Introduced the `-j`/`--json` flag to enable output in JSON format:

```
     Running `target/debug/hermes -j query channel end ibc-0 transfer channel-0`
{"timestamp":"Apr 07 21:48:04.410","level":"INFO","fields":{"message":"Using default configuration from: '.hermes/config.toml'"},"target":"ibc_relayer_cli::commands"}
{"timestamp":"Apr 07 21:48:04.415","level":"INFO","fields":{"message":"Options QueryChannelEndCmd { chain_id: ChainId { id: \"ibc-0\", version: 0 }, port_id: PortId(\"transfer\"), channel_id: ChannelId(\"channel-0\"), height: None }"},"target":"ibc_relayer_cli::commands::query::channel"}
{"status":"success","result":{"connection_hops":["connection-0"],"ordering":"Unordered","remote":{"channel_id":"channel-0","port_id":"transfer"},"state":"Open","version":"ics20-1"}}
```

versus pretty-printing (the default):

```
    Running `target/debug/hermes query channel end ibc-0 transfer channel-0`
Apr 07 21:48:31.510  INFO ibc_relayer_cli::commands: Using default configuration from: '.hermes/config.toml'
Apr 07 21:48:31.514  INFO ibc_relayer_cli::commands::query::channel: Options QueryChannelEndCmd { chain_id: ChainId { id: "ibc-0", version: 0 }, port_id: PortId("transfer"), channel_id: ChannelId("channel-0"), height: None }
Success
{
  "connection_hops": [
    "connection-0"
  ],
  "ordering": "Unordered",
  "remote": {
    "channel_id": "channel-0",
    "port_id": "transfer"
  },
  "state": "Open",
  "version": "ics20-1"
}
```

Notable code changes in this PR:

- override the default `abscissa::EntryPoint` with out custom `EntryPoint` in `relayer-cli/src/entry.rs` [per suggestion](https://github.com/informalsystems/ibc-rs/pull/796#issuecomment-814884114);
    - this allowed us to introduce the global application flag `-j` or `--json` to enable JSON only output (default is pretty printing)
    - also allowed us to remove the global `-v` verbose parameter
- introduced the `json()` method in `conclude.rs` to peek into the application state and infer if JSON output should be on or off
- we can dispense with `relayer-cli/build.rs`
    - this was a workaround introduced in https://github.com/informalsystems/ibc-rs/issues/590
    - the purpose was to ensure that Hermes CLI output shows `hermes` (the binary name) instead of `ibc-relayer-cli` (the crate name)

______

For contributor use:

- [X] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [ ] If applicable: Unit tests written, added test to CI.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

**Comment by romac** [comment]
imho whether or not we output JSON should be a global command-line flag (available on every commands) rather than a config option, otherwise if one wants to use the CLI programmatically (either in a script or in the e2e test suite) one has to ensure that the config file contains the proper value for `log_json`.

I wonder if we could do this by defining our own [`Entrypoint` struct](https://docs.rs/abscissa_core/0.5.2/src/abscissa_core/command/entrypoint.rs.html#11-33, adding the option there, and [telling the framework about it](https://github.com/informalsystems/ibc-rs/blob/6331c5cf01f35a41d0b8e052bec9496048d88f2d/relayer-cli/src/application.rs#L60)?

File: relayer-cli/Cargo.toml [hunk 1]
```diff
@@ -13,7 +13,6 @@ authors = [
 
 description = """
     Implementation of `hermes`, an IBC Relayer developed in Rust.
-    This crate is a CLI wrapper over the `ibc-relayer` library.
```
Comment (romac) [inline relayer-cli/Cargo.toml:16]: 👍

**Comment by romac** [comment]
Looks great, thanks @adizere!

One last thing I wonder now that I am looking at an example: in the human-readable case (ie. when `--json` is not supplied), how about using the standard `dbg!`/`Debug` output instead of still outputting JSON? Since is the output meant to be human-readable, I find the `Debug` output to be more readable than JSON.

Just a thought, based on a personal preference. What do you think?

**Review by cezarad — COMMENTED** [review]
The result of command and raw txs is different: E.g., create channel will print "Channel handshake finished for Channel{  ... } " in a different style than the "Success" message.

**Comment by adizere** [comment]
Status of this PR as of commit a35a225:
- the last commit enforces filtering by modules, so that only log from crates `ibc_relayer` and `ibc_relayer_cli` are now printed
- with this, we can use `debug!` or `trace!` to print lower-level tracing information (without being overwhelmed by tendermint or hyper debug lines)
- this filter is enabled both for JSON output and pretty-printing

The feature of having debug and trace logging appeared to be useful in debugging recent work (https://github.com/informalsystems/ibc-rs/pull/747).

**Review by romac — APPROVED** [review]

**Pull request merged**